tree/ntuple/v7/doc/specifications.md

```diff
@@ -317,20 +317,27 @@ The column type and bits on storage integers can have one of the following value
 | 0x07 |   64 | Real64       | IEEE-754 double precision float                                               |
 | 0x08 |   32 | Real32       | IEEE-754 single precision float                                               |
 | 0x09 |   16 | Real16       | IEEE-754 half precision float                                                 |
-| 0x0A |   64 | Int64        | Two's complement, little-endian 8 byte integer                                |
-| 0x0B |   32 | Int32        | Two's complement, little-endian 4 byte integer                                |
-| 0x0C |   16 | Int16        | Two's complement, little-endian 2 byte integer                                |
-| 0x0D |    8 | Int8         | Two's complement, 1 byte integer                                              |
+| 0x16 |   64 | Int64        | Two's complement, little-endian 8 byte signed integer                         |
+| 0x0A |   64 | UInt64       | Two's complement, little-endian 8 byte unsigned integer                       |
+| 0x17 |   32 | Int32        | Two's complement, little-endian 4 byte signed integer                         |
+| 0x0B |   32 | UInt32       | Two's complement, little-endian 4 byte unsigned integer                       |
+| 0x18 |   16 | Int16        | Two's complement, little-endian 2 byte signed integer                         |
+| 0x0C |   16 | UInt16       | Two's complement, little-endian 2 byte unsigned integer                       |
+| 0x19 |    8 | Int8         | Two's complement, 1 byte signed integer                                       |
+| 0x0D |    8 | UInt8        | Two's complement, 1 byte unsigned integer                                     |
 | 0x0E |   64 | SplitIndex64 | Like Index64 but pages are stored in split + delta encoding                   |
 | 0x0F |   32 | SplitIndex32 | Like Index32 but pages are stored in split + delta encoding                   |
 | 0x10 |   64 | SplitReal64  | Like Real64 but in split encoding                                             |
 | 0x11 |   32 | SplitReal32  | Like Real32 but in split encoding                                             |
 | 0x12 |   16 | SplitReal16  | Like Real16 but in split encoding                                             |
-| 0x13 |   64 | SplitInt64   | Like Int64 but in split encoding                                              |
-| 0x14 |   32 | SplitInt32   | Like Int32 but in split encoding                                              |
-| 0x15 |   16 | SplitInt16   | Like Int16 but in split encoding                                              |
-
-Future versions of the file format may introduce addtional column types
+| 0x1A |   64 | SplitInt64   | Like Int64 but in split encoding                                              |
```

Comment:
It should be the other way round.  Because the currently used `0x13` is now `SplitUInt64`, existing data is only unsplit (no zigzag decoding). Since uint64 can be cast to both `int64_t` and `uint64_t` the data remains readable.

Reply:
Yes, exactly; thanks for confirming :slightly_smiling_face:!